x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authz/permission/FieldPermissions.java

```diff
@@ -43,8 +59,68 @@
  * 2. it must not match the patterns in deniedFieldsArray
  */
 public final class FieldPermissions implements Accountable, CacheKey {
-
     public static final FieldPermissions DEFAULT = new FieldPermissions();
+    // public for testing
+    public static final Set<String> METADATA_FIELDS_ALLOWLIST = Set.of(
+        // built-in
+        IgnoredFieldMapper.NAME,
+        IdFieldMapper.NAME,
+        RoutingFieldMapper.NAME,
+        TimeSeriesIdFieldMapper.NAME,
+        TimeSeriesRoutingHashFieldMapper.NAME,
+        IndexFieldMapper.NAME,
+        IndexModeFieldMapper.NAME,
+        SourceFieldMapper.NAME,
+        IgnoredSourceFieldMapper.NAME,
+        NestedPathFieldMapper.NAME,
+        NestedPathFieldMapper.NAME_PRE_V8,
+        VersionFieldMapper.NAME,
+        SeqNoFieldMapper.NAME,
+        SeqNoFieldMapper.PRIMARY_TERM_NAME,
+        DocCountFieldMapper.NAME,
+        DataStreamTimestampFieldMapper.NAME,
+        FieldNamesFieldMapper.NAME,
+        // plugins
+        // MapperSizePlugin
+        "_size",
+        // MapperExtrasPlugin
+        "_feature",
+        // XPackPlugin
+        DataTierFieldMapper.NAME
+    );
+    public static final FieldPredicate PREDICATE = new FieldPredicate() {
+        @Override
+        public boolean test(String field) {
+            return METADATA_FIELDS_ALLOWLIST.contains(field) || (field.startsWith("_") && startsWithMetadataFieldAndDot(field));
+        }
+
+        /**
+         * Matches {metadata_field}.*
+         */
+        private boolean startsWithMetadataFieldAndDot(String field) {
```

Comment:
This for sub field under allowed metadata fields?

Reply:
Yup! I had a complete naming blank here -- can rename to isSubfield or similar

Reply:
👍 `isMetadataSubField(...)` is I think a more descriptive name.